sys/net/include/netdev.h

```diff
@@ -0,0 +1,271 @@
+                                         either PAN-centric 16-bit address or
+                                         EUI-64 in IEEE 802.15.4) */
+    NETDEV_OPT_EN_PRELOADING,       /**< Enable pre-loading of data, transfer
+                                         data to device using send_data(), send
+                                         by calling trigger(NETDEV_ACTION_TX) */
+    NETDEV_OPT_EN_PROMISCUOUSMODE   /**< Enable promiscuous mode */
+    NETDEV_OPT_EN_AUTOACK,          /**< Automatically send link-layer ACKs */
+    NETDEV_OPT_RSSI,                /**< Read the RSSI value from the last transfer */
+    NETDEV_OPT_LQI,                 /**< Read the link quality indicator */
+
+    /**
+     * @brief   Last value for @ref netdev_opt_t defined here
+     *
+     * @details Specific devices or modules like @ref netapi that utilize these
+     *          values to may define higher values, but they must be greater
+     *          or equal to @ref NETDEV_OPT_LAST.
+     */
+    NETDEV_OPT_LAST,
+} netdev_opt_t;
+
+/**
+ * @brief   Definition of basic network device.
+ * @note    Feel free to expand if your device needs/supports more
+ */
+typedef enum {
+    NETDEV_ACTION_POWEROFF = 0,
+    NETDEV_ACTION_SLEEP,
+    NETDEV_ACTION_IDLE,
+    NETDEV_ACTION_RX,
+    NETDEV_ACTION_TX,               /**< only needed if pre-loading is enabled */
+} netdev_action_t;
+
+/**
+ * TODO: document...
+ */
+typedef enum {
+    NETDEV_EVENT_RX_STARTED = 0,
+    NETDEV_EVENT_RX_COMPLETE,
+    NETDEV_EVENT_TX_STARTED,
+    NETDEV_EVENT_TX_COMPLETE
+    /* probably more? */
+} netdev_event_t;
+
+/**
+ * @brief   Definition of basic network device.
+ * @note    Feel free to expand if your device needs/supports more
+ */
+typedef enum {
+    NETDEV_STATE_OFF = 0,           /**< Device is powered off */
+    NETDEV_STATE_SLEEP,             /**< Device is sleeping */
+    NETDEV_STATE_IDLE,              /**< Device is idle */
+    NETDEV_STATE_RX,                /**< Device is in receive mode */
+    NETDEV_STATE_TX,                /**< Device is transmit mode */
+    NETDEV_STATE_PROMISCUOUS,       /**< Device is in receive mode and
+                                         accepts all packets without regard
+                                         for their destination */
+    NETDEV_STATE_TX_BURST,          /**< Device is burst sending and
+                                         does not accept packets */
+} netdev_state_t;
+
+/**
+ * @brief   The netdev data-structure holds the minimum information needed for
+ *          interaction with MAC layers
+ *
+ * The netdev structure is the parent for all network device driver descriptors.
+ */
+typedef struct netdev_t netdev_t;
+
+/**
+ * @brief   Event callback for signaling event to a MAC layer.
+ *
+ * @param[in] type          type of the event
+ * @param[in] arg           event argument, can e.g. contain a pkt_t pointer
+ */
+typedef void (*netdev_event_cb_t)(netdev_event_t type, void *arg);
```

Comment:
(the structure that comes into `arg` for what event should be well documented is what I want to say ;))